### ISSUE TITLE
MIBOX4 aka MIBOX S exoplayer error when change view visibility

### ISSUE THREAD

**Issue by fgl27** [google/ExoPlayer#8427]
I have open this issue https://github.com/google/ExoPlayer/issues/7996 the problem persist, I can't update it anymore.

Share the apk as was asked, but didn't receive any feedback, was the problem reported to the manufacturer of the device?

I update the source and the apk to test the problem here is it.

The apk

https://drive.google.com/file/d/1eoy3c_TPDxfxwBw6Q4_qsxkJgQOkVZT_/view?usp=sharing

The source repo

https://github.com/fgl27/ExoPlayer/commits/dev-v2_mi_test

The source zip

https://drive.google.com/file/d/1w1of4t1nis6n3O9vZvCkPwfAnShYf7_h/view?usp=sharing

Here is the change

https://github.com/fgl27/ExoPlayer/commit/1cdfba779cfb8403850c410655e00fa1b951698e

This was the head of ExoPlayer dev-v2 branch on the day of the above change

https://github.com/google/ExoPlayer/commits/9753c3fcfb2e2e7fd878666c3f8aa11cbee9728f

Here is the result after 10 sec of playback

![image](https://user-images.githubusercontent.com/13755892/103610511-22a81580-4eff-11eb-99a7-2c8d02a3f654.png)

Here is a demonstration on how to use the app and that it works, at least on the emulator on the MIBOX4 will cause the above error

https://www.youtube.com/watch?v=2RUnSk1nlNo

Basically what that app does is change the playerView visibility to GONE and back to VISIBLE after 10 second of Player.STATE_READY, with causes a playback error on the device.

**Comment by ojw28** [comment]
Thanks for sharing the project! I am able to reproduce the issue. It actually looks like the same underlying problem causes these devices to fail ExoPlayer's test suite, as [documented here](https://exoplayer.dev/oems.html), so this is probably an easier thing to report to the manufacturer of the device. I will do that internally.

**Comment by ojw28** [comment]
@icbaker - We have a bit of a problem, in that these devices don't fail ExoPlayer's test suite any more. This is because they've been added to `MediaCodecVideoRenderer.evaluateDeviceNeedsSetOutputSurfaceWorkaround`. I can't remember if we discussed this already, but I think we need a way of disabling workarounds for the GTS tests. Perhaps we should use it with caution initially, but disabling the output surface workaround seems like a good thing to start with.

Could we do this first, so that any instructions I provide to the OEM will definitely still be valid?

**Comment by icbaker** [comment]
The `setOutputSurface` workaround is now disabled in the GTS tests on the dev-v2 branch, so this device ought to now fail (I don't have a device to test with). Assigning back to @ojw28 to reach out to the OEM.

**Comment by ojw28** [comment]
Reported to Xiaomi in [Internal ref: b/179940201]. Leaving this issue open for tracking.

**Comment by ojw28** [comment]
I don't have any details regarding their schedule, but Xiaomi have a fix for the issue that they plan on releasing to users in an upcoming software update. Closing as fixed on this basis.

We will also make sure that we have good coverage for this in our device certification suite going forward (@icbaker - I think it's on your radar already anyway, but just FYI this is a good reason for updating GTS!).
